src/dysh/spectra/spectrum.py

```diff
@@ -129,30 +155,60 @@ def baseline(self, degree, exclude=None, **kwargs):
 
                 Default: no exclude region
 
+            include: list of 2-tuples of int (currently units not supported yet, pending issue 251/260)
+
             model : str
-                One of 'polynomial' or 'chebyshev', Default: 'polynomial'
+                One of 'polynomial' 'chebyshev', 'legendre', or 'hermite'
+                Default: 'polynomial'
             fitter  :  `~astropy.fitting._FitterMeta`
                 The fitter to use. Default: `~astropy.fitter.LinearLSQFitter` (with `calc_uncertaintes=True`).
                 Be care when choosing a different fitter to be sure it is optimized for this problem.
             remove : bool
                 If True, the baseline is removed from the spectrum. Default: False
+            normalize : bool
+                If True, the frequency axis is internally rescaled from 0..nchan-1
+                to avoid roundoff problems (and make the coefficients slightly more
+                understandable). This is usually needed for a polynomial, though overkill
+                for the others who do their own normalization.
+                @todo maybe allow True, False, None; the latter will use True for polynamical, False for others
+                Default: True
 
         """
         # fmt: on
         # @todo: Are exclusion regions OR'd with the existing mask? make that an option?
         kwargs_opts = {
             "remove": False,
+            "normalize": False,
             "model": "polynomial",
             "fitter": LinearLSQFitter(calc_uncertainties=True),
         }
         kwargs_opts.update(kwargs)
 
+        if kwargs_opts["normalize"]:
+            print("Warning:  baseline fit done in channel space")
+            spectral_axis = np.copy(self._spectral_axis.value)
+            nchan = len(spectral_axis)
+            # sadly, there is no single setter for this
+            for i in range(nchan):
+                self._spectral_axis[i] = i * u.Hz  # would like to use "chan" units
+            self._normalized = True
+            # allow include
+            if include != None:
+                nchan = len(spectral_axis)
+                exclude = self._toggle_sections(nchan, include)
+
         self._baseline_model = baseline(self, degree, exclude, **kwargs)
+
         if kwargs_opts["remove"]:
             s = self.subtract(self._baseline_model(self.spectral_axis))
             self._data = s._data
             self._subtracted = True
 
+        if kwargs_opts["normalize"]:
+            for i in range(nchan):
+                self._spectral_axis[i] = spectral_axis[i] * u.Hz
```

Comment:
Line 209: What if the original unit wasn't Hz?    Need to save the unit or deep copy the entire spectral axis and restore it with self._spectral_axis = saved_spectral_axis